pydeseq2/utils.py

```diff
@@ -1520,3 +1522,73 @@ def make_MA_plot(
 
     if save_path is not None:
         plt.savefig(save_path, bbox_inches="tight")
+
+
+# Authors: Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#
+# License: BSD (3-clause)
+
+# Adapted from https://gist.github.com/agramfort/850437
+
+
+def lowess(
+    features: np.ndarray, targets: np.ndarray, frac: float = 2.0 / 3.0, iter: int = 3
+):
+    """Run lowess smoothing: Robust locally weighted regression.
+
+    The lowess function fits a nonparametric regression curve to a scatterplot.
+    The arrays features and targets contain an equal number of elements; each pair
+    (features[i], targets[i]) defines a data point in the scatterplot. The function
+    returns the estimated (smooth) values of targets.
+    The smoothing span is given by f. A larger value for f will result in a
+    smoother curve. The number of robustifying iterations is given by iter. The
+    function will run faster with a smaller number of iterations.
+
+    Parameters
+    ----------
+    features : ndarray
+        Data points.
+    targets : ndarray
+        Target values.
+    frac : float
+        The fraction of the data used when estimating each y-value. (default: ``2/3``).
+    iter : int
+        The number of robustifying iterations. (default: ``3``).
+
+    Returns
+    -------
+    ndarray
+        Estimated (smooth) values of targets.
+    """
+    n = len(features)
+    r = int(ceil(frac * n))
+    h = np.maximum(
+        np.array([np.sort(np.abs(features - features[i]))[r] for i in range(n)]), 1e-12
+    )
+    w = np.clip(
+        np.abs(np.nan_to_num((features[:, None] - features[None, :]) / h)), 0.0, 1.0
```

Comment:
maybe for coherence put h[:,None]

Reply:
Do you think it would make the code clearer? Given that the code is working as is, I'm a bit reluctant to add unnecessary broadcasting operations.